### PR TITLE
fix(RHTAPBUGS-763): run lint checks for prs

### PR DESCRIPTION
**Pull request by Troy876**
Adds the workflow file to run lint checks for new PRs
Also adds the CODEOWNERS file

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# This is a comment.
```
Comment (davidmogar) [inline .github/CODEOWNERS:1]: This change is unrelated. The PR should contain only linting changes.

**Review by davidmogar — DISMISSED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (johnbieren) [reply .github/CODEOWNERS:1]: I asked him to add this. @Troy876 can you open a second PR to add the CODEOWNERS instead of including it in this one?

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+name: Linters
+'on':
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - synchronize
+  workflow_dispatch: null
```
Comment (johnbieren) [inline .github/workflows/lint.yaml:11]: can you remove this line?

File: .github/CODEOWNERS (hunk 1 above)
Comment (Troy876) [reply .github/CODEOWNERS:1]: Sure thing

**Review by johnbieren — COMMENTED** [review]
I believe https://github.com/redhat-appstudio/release-service/blob/main/.github/gitlint/contrib_format_conventional_commits.py should be added to for gitlint to work properly

**Comment by Troy876** [comment]
> I believe https://github.com/redhat-appstudio/release-service/blob/main/.github/gitlint/contrib_format_conventional_commits.py should be added to for gitlint to work properly

I believe that has already been added in #55

**Comment by johnbieren** [comment]
> > I believe https://github.com/redhat-appstudio/release-service/blob/main/.github/gitlint/contrib_format_conventional_commits.py should be added to for gitlint to work properly
> 
> I believe that has already been added in #55

Ah, my bad!

**Review by johnbieren — APPROVED** [review]
